pallets/author-mapping/src/migrations.rs

```diff
@@ -14,20 +14,73 @@
 // You should have received a copy of the GNU General Public License
 // along with Moonbeam.  If not, see <http://www.gnu.org/licenses/>.
 
-use crate::{BalanceOf, Config, MappingWithDeposit, RegistrationInfo};
+use crate::{
+	BalanceOf, Config, MappingWithDeposit, MappingWithDepositAndKeys, RegistrationInfo,
+	RegistrationInformation,
+};
 use frame_support::{
 	pallet_prelude::PhantomData,
 	storage::migration::{remove_storage_prefix, storage_key_iter},
 	traits::{Get, OnRuntimeUpgrade},
 	weights::Weight,
-	Twox64Concat,
+	Blake2_128Concat, Twox64Concat,
 };
 use nimbus_primitives::NimbusId;
 
 use sp_std::convert::TryInto;
 //TODO sometimes this is unused, sometimes its necessary
 use sp_std::vec::Vec;
 
+// TODO: configure in runtime migrations
+/// Migrates MappingWithDeposit map value from RegistrationInfo to RegistrationInformation,
+/// thereby adding a keys: T::Keys field to the value to support VRF keys that can be looked up
+/// via NimbusId.
+pub struct AddKeysToRegistrationInfo<T>(PhantomData<T>);
+impl<T: Config> OnRuntimeUpgrade for AddKeysToRegistrationInfo<T> {
+	fn on_runtime_upgrade() -> Weight {
+		log::info!(target: "AddKeysToRegistrationInfo", "actually running it");
+		let pallet_prefix: &[u8] = b"AuthorMapping";
+		let old_storage_item_prefix: &[u8] = b"MappingWithDeposit";
+
+		// Read all the data into memory.
+		// https://crates.parity.io/frame_support/storage/migration/fn.storage_key_iter.html
+		let stored_data: Vec<_> = storage_key_iter::<
```

Comment:
The cleanest way is to keep the same storage item and use translate to migrate data : https://paritytech.github.io/substrate/master/frame_support/storage/trait.IterableStorageMap.html#tymethod.translate

So, it implies to modify directly the existing type `RegistrattionInfo` in the pallet, and create a private type `OldRegistrattionInfo` in migrations module to use in the migration only.